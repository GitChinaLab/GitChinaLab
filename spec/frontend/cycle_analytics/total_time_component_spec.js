import { mount } from '@vue/test-utils';
import TotalTimeComponent from '~/cycle_analytics/components/total_time_component.vue';

describe('TotalTimeComponent', () => {
  let wrapper = null;

  const createComponent = (propsData) => {
    return mount(TotalTimeComponent, {
      propsData,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('with a valid time object', () => {
    it.each`
      time
      ${{ seconds: 35 }}
      ${{ mins: 47, seconds: 3 }}
      ${{ days: 3, mins: 47, seconds: 3 }}
      ${{ hours: 23, mins: 10 }}
      ${{ hours: 7, mins: 20, seconds: 10 }}
    `('with $time', ({ time }) => {
      wrapper = createComponent({
        time,
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('with a blank object', () => {
    beforeEach(() => {
      wrapper = createComponent({
        time: {},
      });
    });

    it('should render --', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
