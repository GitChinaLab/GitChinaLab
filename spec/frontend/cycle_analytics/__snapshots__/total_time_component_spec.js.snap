// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalTimeComponent with a blank object should render -- 1`] = `"<span class=\\"total-time\\"> -- </span>"`;

exports[`TotalTimeComponent with a valid time object with {"days": 3, "mins": 47, "seconds": 3} 1`] = `
"<span class=\\"total-time\\">
    3 <span>days</span></span>"
`;

exports[`TotalTimeComponent with a valid time object with {"hours": 7, "mins": 20, "seconds": 10} 1`] = `
"<span class=\\"total-time\\">
    7 <span>hrs</span></span>"
`;

exports[`TotalTimeComponent with a valid time object with {"hours": 23, "mins": 10} 1`] = `
"<span class=\\"total-time\\">
    23 <span>hrs</span></span>"
`;

exports[`TotalTimeComponent with a valid time object with {"mins": 47, "seconds": 3} 1`] = `
"<span class=\\"total-time\\">
    47 <span>mins</span></span>"
`;

exports[`TotalTimeComponent with a valid time object with {"seconds": 35} 1`] = `
"<span class=\\"total-time\\">
    35 <span>s</span></span>"
`;
